File: lib/models/data/scheme.dart
import 'package:flutter_terminal_sdk/models/data/pos_host_details.dart';

import 'label_value.dart';

class Scheme {
  final LabelValue<String> name;
  final PosHostDetails pos;
  final PosHostDetails host;
  final bool isBalanced;

  Scheme({
    required this.name,
    required this.pos,
    required this.host,
    required this.isBalanced,
  });

  factory Scheme.fromJson(Map<String, dynamic> json) {
    return Scheme(
      name: LabelValue<String>.fromJson(json['name'] as Map<String, dynamic>),
      pos: PosHostDetails.fromJson(json['pos'] as Map<String, dynamic>),
      host: PosHostDetails.fromJson(json['host'] as Map<String, dynamic>),
      isBalanced: json['isBalanced'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name.toJson(),
      'pos': pos.toJson(),
      'host': host.toJson(),
      'isBalanced': isBalanced,
    };
  }
}

File: android/terminalSdk/src/main/java/io/nearpay/terminalsdk/data/dto/Scheme.kt
package io.nearpay.terminalsdk.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Scheme(
    @SerialName("name")
    val name: LabelValue<String>,

    @SerialName("pos")
    val pos: PosHostDetails,

    @SerialName("host")
    val host: PosHostDetails,

    @SerialName("is_balanced")
    val isBalanced: Boolean
)